collections/take_last_while.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+/**
+ * Returns all elements in the given collection until the first element that does not match the given predicate.
+ *
+ * Example:
+ * ```ts
+ * import { takeLastWhile } from "./take_last_while.ts";
+ * import { assertEquals } from "../testing/asserts.ts";
+ *
+ * const arr = [1, 2, 3, 4, 5, 6];
+ *
+ * assertEquals(
+ *   takeLastWhile(arr, (i) => i > 4),
+ *   [5, 6],
+ * );
+ * ```
+ */
+export function takeLastWhile<T>(
+  array: readonly T[],
+  predicate: (el: T) => boolean,
+): T[] {
+  const newArray: T[] = [];
+  const length = array.length;
+
+  for (let i = length - 1; i >= 0; i--) {
+    if (predicate(array[i])) {
+      newArray.push(array[i]);
+    } else {
+      break;
+    }
+  }
+
+  return newArray.reverse();
```

Comment:
```suggestion
  let offset = array.length;
  while (0 < offset && predicate(array[offset - 1])) offset--;

  return array.slice(offset, array.length);
```

Reply:
Thanks for this!

Reply:
Isn't

```ts
let offset = array.length;
while (0 < offset && predicate(array[offset - 1])) offset--;
```

pretty much `findLastIndex()`, which we already have implemented in the module?

Reply:
Update:

Doing some spontanous benchmarking, I can not find any significant stable performance difference between

```ts
  const index = findLastIndex(array, (it) => !predicate(it))

  if (index === undefined) {
      return []
  }

  return array.slice(index + 1);
```

and

```ts
  let offset = array.length;
  while (0 < offset && predicate(array[offset - 1])) offset--;

  return array.slice(offset, array.length);
```

I guess either is fine, I just want to emphasize that if we can DRY code up without a significant impact, we probably should for maintenance reasons.